proposed/2022/dotnet-nuget-config-spec.md

```diff
@@ -0,0 +1,358 @@
+
+Please note this command only manages settings in [config section](https://learn.microsoft.com/en-us/nuget/reference/nuget-config-file#config-section).
+For other settings not in config section, we have/will have other commands. E.g. for [trustedSigners section](https://learn.microsoft.com/en-us/nuget/reference/nuget-config-file#trustedsigners-section), we have [dotnet nuget trust](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-nuget-trust) command.
+
+#### Arguments
+
+- CONFIG_KEY
+
+Specify the key of the settings that are to be removed.
+
+If the specified `CONFIG_KEY` is not a key in [config section](https://learn.microsoft.com/en-us/nuget/reference/nuget-config-file#config-section), a message is displayed indicating the `CONFIG_KEY` could not be unset.
+
+#### Options
+
+- --config-file
+
+Specify the config file path to remove the setting key-value pair. If it's not specified, `%AppData%\NuGet\NuGet.Config` (Windows), or `~/.nuget/NuGet/NuGet.Config` or `~/.config/NuGet/NuGet.Config` (Mac/Linux) is used. See [On Mac/Linux, the user-level config file location varies by tooling.](https://learn.microsoft.com/en-us/nuget/consume-packages/configuring-nuget-behavior#on-maclinux-the-user-level-config-file-location-varies-by-tooling)
+
+- -?|-h|--help
+
+Prints out a description of how to use the command.
+
+#### Examples
+
+- Unset the `signatureValidationMode` in the user-wide NuGet configuration file.
+
+```
+dotnet nuget config unset signatureValidationMode
+
+```
+
+- Unset the `defaultPushSource` in the specified NuGet configuration file.
+
+```
+dotnet nuget config unset defaultPushSource --config-file C:\Users\username\AppData\Roaming\NuGet\NuGet.Config
+
+```
+
+## Future Work
+1. The `dotnet nuget config path/get` is a community ask. We will discuss if adding this command into NuGet.exe CLI, in the future.
+
+2. NuGet.exe [config command](https://learn.microsoft.com/en-us/nuget/reference/cli-reference/cli-ref-config) is implemented. 
+But the behavior is confusing: the `set` command will set the property which appears last when loading(for all writable files), so it's not updating the closest NuGet configuration file, but the user-wide NuGet configuration file.(Related code: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs#L229)
+We keep the behavior of `dotnet nuget config set` the same with above for now.
+But we might need to change this behavior in the future. 
+Considering this will be a breaking change, we will only consider doing it in main version change, if there are enough votes.
+
+## Open Questions
+1. To show configuration files locations, is it better to use `--verbosity` option or some option named like `--show-path`? (git command is using `--show-origin` for similar purpose)
+
+2. `dotnet nuget config get <CONFIG_KEY>` will get the value of the specifc config key (which is aligned with NuGet config command). But
```

Comment:
I think it's ok if `dotnet nuget config get` shows all the config settings, I just don't think we should show the XML.

Reply:
Thanks! Changed the format accordingly. We might need to go through all configuration sections for the formats later.